src/OrchardCore/OrchardCore.Infrastructure/Scripting/CommonGeneratorMethods.cs

```diff
@@ -28,26 +36,46 @@ public class CommonGeneratorMethods : IGlobalMethodProvider
     /// Converts a Base64 encoded gzip stream to an uncompressed Base64 string.
     /// See http://www.txtwizard.net/compression.
     /// </summary>
-    private static readonly GlobalMethod _gZip = new()
+    private readonly GlobalMethod _gZip = new()
     {
         Name = "gzip",
         Method = serviceProvider => (Func<string, string>)(encoded =>
         {
-            var bytes = Convert.FromBase64String(encoded);
-            using var gzip = new GZipStream(new MemoryStream(bytes), CompressionMode.Decompress);
+            var neededLength = GetByteArrayLengthFromBase64(encoded);
+
+            Span<byte> bytes = new byte[neededLength];
+
+            if (!Convert.TryFromBase64String(encoded, bytes, out var _))
+            {
+                throw new FormatException("Invalid Base64 string.");
+            }
+
+            using var stream = MemoryStreamFactory.GetStream();
```

Comment:
And when using `GetSpan` this becomes unnecessary because the conversion is done directly on the stream.